### ISSUE TITLE
New/changed flight plans not recognized

### ISSUE THREAD

**Issue by droesen** [qutescoop/qutescoop#51]
Hi,

a long-standing bug I'm observing is that QuteScoop often does not notice a flight plan change for a client that is online, e.g. when doing a quick turnaround at an airport, submitting a new flight plan under the same callsign as before, without going offline. But I'm not sure that this is the only situation where I'm seeing it. Repeated reloading of data via F5 doesn't help, I need to restart QuteScoop to get the current FPL. I will try to record further incidents with the associated details to this issue report - just wanted to get the problem "on record" for now as it's the most annoying bug for me. :-)

Best regards,
Daniel

**Comment by jonaseberle** [comment]
Hi Daniel, thanks for reporting that. 

I can confirm. 
Even when logging in without FP, then sending one via https://my.vatsim.net/pilots/flightplan:
The data is properly in https://data.vatsim.net/v3/vatsim-data.json and the updated data is generally loaded by QuteScoop, but the `Pilot` flightplan does not get updated.

**Comment by jonaseberle** [comment]
I debugged a bit.
Not only the flightplan, I get no updated values at all. I found the easiest to test was the transponder code.

Looking here https://data.vatsim.net/v3/vatsim-data.json
or using `curl --silent https://data.vatsim.net/v3/vatsim-data.json | jq -r '.pilots[] | select(.cid == <your VATSIM ID>).transponder'` to get my transponder code I traced it:

* in `WhazzupData` - ok
* in `Pilot` - ok
* but then the previous data is still here: `GLWidget::createPilotsList()`
This uses `Whazzup::instance()->whazzupData().pilots.values()` to get the pilots so this does not seem update properly.

...

I think I found the culprit!

In https://github.com/qutescoop/qutescoop/blob/master/src/WhazzupData.cpp#L401-L404

it only updates the data if the plane has moved!
That's why your FP wasn't updated while you were at the gate and not moving.

What a nasty little bug ;)
